xpdan/callbacks.py

```diff
@@ -6,4 +6,4 @@ def start(self, doc):
         print('START ANALYSIS ON {}'.format(doc['uid']))
 
     def stop(self, doc):
-        print('FINISH ANALYSIS ON {}'.format(doc['run_start']))
+        print('FINISH ANALYSIS ON {}'.format(doc.get('run_start', 'NA')))
```

Comment:
is this a warning or error message?  Will it be clear to users what it means (or if they cut-paste it into ``xpd-users`` can be used for debugging as is?).   Is this a regular stop or an irregular stop?  For a regular stop it could say something like "pipeline completed analysis normally\n Analysis finished on {}.format..."

If it is used as a warning, something like:
"analysis terminated prematurely\n 
Analysis finished on {}'.form....\n
Cut and paste this message into xpd-users for trouble-shooting"

or sthg.

Reply:
This message was initially intended as a kind of heartbeat for the pipeline, letting users know if it is currently processing data or if it has finished (and which data it finished on). I have had users send me this so I could figure out what was going on, and it does help users know how much data there is in the queue.

Formally it isn't a problem for the pipeline if a stop isn't issued (at least when this PR is merged and it won't keep crashing the pipeline), but it is rather odd for the run engine to not issue a stop (we've been running into this about once or twice a beamtime which is a little bit disconcerting). So I wouldn't do anything to alarm the users, but we do need to handle it somehow.